README.md

```diff
@@ -1,15 +1,122 @@
+/home/mgoerens/dev/github.com/opdev/synapse-operator/bin/kustomize build config/crd | kubectl apply -f -
+customresourcedefinition.apiextensions.k8s.io/synapses.synapse.opdev.io configured
+```
+
+Run the controller locally with:
+
+```shell
+$ make run
+/home/mgoerens/dev/github.com/opdev/synapse-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+/home/mgoerens/dev/github.com/opdev/synapse-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+go fmt ./...
+go vet ./...
+go run ./main.go
+I0329 13:14:32.483664   26252 request.go:665] Waited for 1.039947837s due to client-side throttling, not priority and fairness, request: GET:https://api.crc.testing:6443/apis/operators.coreos.com/v1?timeout=32s
+1.6485524737848275e+09	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": ":8080"}
+1.648552473785254e+09	INFO	setup	starting manager
+1.6485524737853944e+09	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
+1.6485524737853956e+09	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
+1.6485524737854843e+09	INFO	controller.synapse	Starting EventSource	{"reconciler group": "synapse.opdev.io", "reconciler kind": "Synapse", "source": "kind source: *v1alpha1.Synapse"}
+1.648552473785519e+09	INFO	controller.synapse	Starting EventSource	{"reconciler group": "synapse.opdev.io", "reconciler kind": "Synapse", "source": "kind source: *v1.Service"}
+1.6485524737855291e+09	INFO	controller.synapse	Starting EventSource	{"reconciler group": "synapse.opdev.io", "reconciler kind": "Synapse", "source": "kind source: *v1.Deployment"}
+1.6485524737855399e+09	INFO	controller.synapse	Starting EventSource	{"reconciler group": "synapse.opdev.io", "reconciler kind": "Synapse", "source": "kind source: *v1.PersistentVolumeClaim"}
+1.6485524737855482e+09	INFO	controller.synapse	Starting Controller	{"reconciler group": "synapse.opdev.io", "reconciler kind": "Synapse"}
+1.6485524738865452e+09	INFO	controller.synapse	Starting workers	{"reconciler group": "synapse.opdev.io", "reconciler kind": "Synapse", "worker count": 1}
+
+```
+
+This runs the controller until you hit `Ctrl` + `C`.
+
+To uninstall the `Synapse `CRD:
+
+```shell
+$ make uninstall
+/home/mgoerens/dev/github.com/opdev/synapse-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+/home/mgoerens/dev/github.com/opdev/synapse-operator/bin/kustomize build config/crd | kubectl delete -f -
+customresourcedefinition.apiextensions.k8s.io "synapses.synapse.opdev.io" deleted
+```
+
+### Deploy the controller in the Kubernetes cluster with `make deploy`
+
+Deploy the controller with:
+
+```shell
+$ make deploy
```

Comment:
Just tried a make deploy like a regular user. I was on the default namespace and run make deploy.
That's what I got:
```
 oc get pods -n synapse-operator-sy
stem
NAME                                                   READY   STATUS             RESTARTS   AGE
synapse-operator-controller-manager-7b967dd8c5-4r2dp   1/2     ImagePullBackOff   0          7m36s
```
```
  Warning  Failed   7m12s (x3 over 7m54s)   kubelet  Error: ErrImagePull
  Warning  Failed   6m35s (x6 over 7m54s)   kubelet  Error: ImagePullBackOff
  Normal   BackOff  2m48s (x21 over 7m54s)  kubelet  Back-off pulling image "controller:latest"
```
Can you confirm it's updating the image name on your setup? I tried this on OpenShift.

Reply:
I have the same issue in my setup. It makes sense though, the container image is not available on the Kubernetes cluster. In fact, even locally, I haven't built a container image for the controller yet.

I believe we need to have a public controller image.

Reply:
Let's merge this PR as-is. As discussed, we just need to build and push public docker images (+ correctly configure the Makefile) for this to work.